### PR TITLE
Next stage of CoNLL scenario

### PR DESCRIPTION
**Pull request by frreiss**
This PR adds several of the planned steps to the CoNLL-2003 scenario (see issue #5):
* Retokenize with a BERT tokenizer,
* Align spans between the BERT and CoNLL tokenizations
* Generates BERT embeddings
* Train a single baseline model over the BERT embeddings
* Train multiple reduced-quality models on the embeddings
* Preliminary examination of the outputs of the models to ensure we still can localize incorrect labels when we generate the multiple models ourselves

Most of the changes are in `CoNLL_3.ipynb`. I've also extended the code in `text_extensions_for_pandas/io/tokenization.py` for aligning spans across tokenizations.

### CODE REVIEW

File: text_extensions_for_pandas/io/tokenization.py [hunk 1]
```diff
@@ -70,28 +74,171 @@ def make_bert_tokens(target_text: str, tokenizer) -> pd.DataFrame:
...
+    if len(seq.shape) != 1:
+        raise ValueError(f"Input array must be 1D; got shape {seq.shape}")
+    window_length, pre_padding, post_padding = _compute_padding(
+        len(seq), overlap, non_overlap
+    )
+
+    # First generate the windows as a padded flat arrays.
+    padded_length = len(seq) + pre_padding + post_padding
+    buf = np.zeros(shape=[padded_length], dtype=seq.dtype)
+    buf[pre_padding : pre_padding + len(seq)] = seq
```
Comment (BryanCutler) [inline text_extensions_for_pandas/io/tokenization.py:137]: would `np.concatenate` be better than making a copy of `seq`?

File: text_extensions_for_pandas/io/test_tokenization.py [hunk 2]
```diff
@@ -76,3 +79,28 @@ def test_make_bert_tokens(self):
...
+
+    def test_seq_to_windows(self):
+        for seqlen in range(1, 20):
+            seq = np.arange(1, seqlen)
+            seq_after = windows_to_seq(seq, seq_to_windows(seq, 2, 3)["input_ids"], 2, 3)
+            if np.any(seq != seq_after):
+                raise ValueError("Before: {seq}; After: {seq_after}")
+
+        for seqlen in range(200, 400):
+            # print(f"seqlen: {seqlen}")
```
Comment (BryanCutler) [inline text_extensions_for_pandas/io/test_tokenization.py:91]: leftover print statements?

File: text_extensions_for_pandas/io/tokenization.py (hunk 1 above)
Comment (frreiss) [reply text_extensions_for_pandas/io/tokenization.py:137]: I don't think there would be any performance benefit to using `np.concatenate` here. As far as I'm aware, `np.concatenate` always allocates a fresh, contiguous buffer and copies the arrays being concatenated into that buffer; see source code at https://github.com/numpy/numpy/blob/50ce0fce70cac779919d97578381a9d762f42594/numpy/core/src/multiarray/multiarraymodule.c#L363

File: text_extensions_for_pandas/io/test_tokenization.py (hunk 2 above)
Comment (frreiss) [reply text_extensions_for_pandas/io/test_tokenization.py]: Good catch! I've removed these statements in dbb7da9

**Pull request merged**